yarn-project/p2p/src/services/reqresp/reqresp.ts

```diff
@@ -450,7 +459,8 @@ export class ReqResp {
 
       // Open the stream with a timeout
       const result = await executeTimeout<ReqRespResponse>(
-        (): Promise<ReqRespResponse> => pipe([payload], stream!, this.readMessage.bind(this)),
+        (signal): Promise<ReqRespResponse> =>
+          pipe([payload], abortableDuplex(stream!, signal), abortableSink(this.readMessage.bind(this), signal)),
         this.individualRequestTimeoutMs,
         () => new IndividualReqRespTimeoutError(),
       );
```

Comment:
Reading more about signals I found `AbortSignal.any`. Maybe we could rewrite this _without_ `executeTimeout`, and instead creating an `AbortSignal.any([AbortSignal.timeout(this.individualRequestTimeoutMs), signal])`?

Reply:
Also, the Charlie in me wonders if we can rewrite this without the extra libs and just using node's `stream` coupled with `addAbortSignal` as shown [here](https://betterstack.com/community/guides/scaling-nodejs/understanding-abortcontroller/#cancelling-streams-with-abortcontroller).

Reply:
I wish we we're using Node's pipeline function here (which natively supports `AbortSignal`) but this is a custom pipe function so I ended up using a package from the same author

Reply:
Yep, I was thinking about refactoring it to the point of dropping `pipe` in favor of node's native stream methods. And yes, it's way out of scope of this PR.